Source/VMware.vSphereDSC/Tests/Unit/VMHostSyslog.Unit.Tests.ps1

```diff
@@ -0,0 +1,985 @@
+                }
+                $setSyslogConfigMock = {
+                    return $true
+                }
+
+                Mock -CommandName Connect-VIServer -MockWith $viServerMock -ModuleName $script:moduleName
+                Mock -CommandName Get-VMHost -MockWith $vmHostMock -ModuleName $script:moduleName
+                Mock -CommandName Get-EsxCli -MockWith $esxCliMock -ModuleName $script:moduleName
+                Mock -CommandName Get-VMHostSyslogConfig -MockWith $getSyslogConfigMock -ModuleName $script:moduleName
+                Mock -CommandName Set-VMHostSyslogConfig -MockWith $setSyslogConfigMock -ModuleName $script:moduleName
+            }
+
+            # Arrange
+            $resource = New-Object -TypeName $script:resourceName -Property $script:resourceProperties
+
+            It 'Should call the Connect-VIServer mock with the passed server and credentials once' {
+                # Act
+                $resource.Get()
+
+                # Assert
+                Assert-MockCalled -CommandName Connect-VIServer `
+                    -ParameterFilter { $Server -eq $script:resourceProperties.Server -and $Credential -eq $script:resourceProperties.Credential } `
+                    -ModuleName $script:moduleName -Exactly 1 -Scope It
+            }
+
+            It 'Should call Get-VMHost mock with the passed server and name once' {
+                # Act
+                $resource.Get()
+
+                # Assert
+                Assert-MockCalled -CommandName Get-VMHost `
+                    -ParameterFilter { $Server -eq $viServer -and $Name -eq $script:resourceProperties.Name } `
+                    -ModuleName $script:moduleName -Exactly 1 -Scope It
+            }
+
+            It 'Should call Get-EsxCli mock once' {
+                # Act
+                $resource.Get()
+
+                # Assert
+                Assert-MockCalled -CommandName Get-EsxCli `
+                    -ParameterFilter { $Server -eq $viServer -and $VMHost -eq $vmhost -and $V2 } `
+                    -ModuleName $script:moduleName -Exactly 1 -Scope It
+            }
+
+            It 'Should call Get-VMHostSyslogConfig mock once' {
+                # Act
+                $resource.Get()
+
+                # Assert
+                Assert-MockCalled -CommandName Get-VMHostSyslogConfig `
+                    -ParameterFilter { $EsxCli.VMHost.Id -eq $VMHost.Id } `
+                    -ModuleName $script:moduleName -Exactly 1 -Scope It
+            }
+
+            It 'Should not call Set-VMHostSyslogConfig mock' {
+                # Act
+                $resource.Get()
+
+                # Assert
+                Assert-MockCalled -CommandName Set-VMHostSyslogConfig `
+                    -ParameterFilter { $EsxCli.VMHost.Id -eq $VMHost.Id -and $VMHostSyslogConfig.LogHost -eq $script:resourceProperties.LogHost  } `
+                    -ModuleName $script:moduleName -Exactly 0 -Scope It
+            }
```

Comment:
I do not see test checking the result returned from the Get() method. Those should also be implemented.